src/plots/waterfallPlot.ts

```diff
@@ -0,0 +1,194 @@
+     * 
+     * @param {boolean|Accessor<number>}
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public total(total: boolean | Accessor<boolean>): Waterfall<X, Y>;
+    /**
+     * Sets total to a scaled constant value or scaled result of an Accessor.
+     * The provided Scale will account for the values when autoDomain()-ing
+     * 
+     * @param {T|Accessor<T>}
+     * @param {Scale<T, number>} scale
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public total<T>(total: boolean | Accessor<T>, scale: Scale<T, boolean>): Waterfall<X, Y>;
+    public total<T>(total?: boolean | Accessor<boolean> | T | Accessor<T>, scale?: Scale<T, boolean>): any {
+      if (total === undefined) {
+        return this._propertyBindings.get(Waterfall._TOTAL_KEY);
+      }
+      this._bindProperty(Waterfall._TOTAL_KEY, total, scale);
+      return this;
+    }
+
+    protected _additionalPaint(time: number) {
+      this._connectorArea.selectAll("line").remove();
+      if (this._connectorsEnabled) {
+        Utils.Window.setTimeout(() => this._drawConnectors(), time);
+      }
+    }
+
+    protected _createNodesForDataset(dataset: Dataset) {
+      var drawer = super._createNodesForDataset(dataset);
+      this._connectorArea = this._renderArea.append("g").classed(Waterfall._CONNECTOR_AREA_CLASS, true);
+      return drawer;
+    }
+
+    protected _extentsForProperty(attr: string) {
+      var primaryAttr = "y";
+      if (attr === primaryAttr) {
+        return [this._extent];
+      } else {
+        return super._extentsForProperty(attr);
+      }
+    }
+
+    protected _generateAttrToProjector() {
+      var attrToProjector = super._generateAttrToProjector();
+
+      var yScale = this.y().scale;
+      var totalAccessor = Plot._scaledAccessor(this.total());
+
+      attrToProjector["y"] = (d, i, dataset) => {
+        var isTotal = totalAccessor(d, i, dataset);
+        if (isTotal) {
+          return Plot._scaledAccessor(this.y())(d, i, dataset);
+        } else {
+          var currentSubtotal = this._subtotals[i];
+          var priorSubtotal = this._subtotals[i - 1];
+          if (currentSubtotal > priorSubtotal) {
+            return yScale.scale(<any> currentSubtotal);
+          } else {
+            return yScale.scale(<any> priorSubtotal);
+          }
+        }
+      };
+
+      attrToProjector["height"] = (d, i, dataset) => {
+        var isTotal = totalAccessor(d, i, dataset);
+        var currentValue = this.y().accessor(d, i, dataset);
+        if (isTotal) {
+          return Math.abs(yScale.scale(<any> currentValue) - yScale.scale(<any> 0));
+        } else {
+          var currentSubtotal = this._subtotals[i];
+          var priorSubtotal = this._subtotals[i - 1];
+          var height = Math.abs(yScale.scale(<any> currentSubtotal) - yScale.scale(<any> priorSubtotal));
+          return height;
+        }
+      };
+
+      attrToProjector["class"] = (d, i, dataset) => {
```

Comment:
Yeah (reported in #1862 from before), need to figure out a way to resolve this one..